metaci/plan/migrations/0040_plan_commit_status_regex.py

```diff
@@ -0,0 +1,35 @@
+# Generated by Django 3.1.13 on 2021-08-09 16:52
+
+from django.db import migrations, models
+
+
+def migrate_regexes(apps, schema_editor):
+    Plan = apps.get_model("plan", "Plan")
+    for plan in Plan.objects.all():
+        if plan.trigger == "status" and plan.regex:
+            # Migrate the regex to the new commit_status_regex field.
+
+            plan.commit_status_regex = plan.regex
+            plan.regex = None
+            plan.save()
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("plan", "0039_plan_change_traffic_control"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="plan",
+            name="commit_status_regex",
+            field=models.CharField(
+                blank=True,
+                help_text="For Plans that use the trigger Commit Status, run builds when the commit status matches this regex.",
+                max_length=255,
+                null=True,
+            ),
+        ),
+        migrations.RunPython(migrate_regexes),
```

Comment:
The lack of a backward migration to copy from commit_status_regex to regex means that we'll lose the regex values if we have to migrate backward (the framework will run the default no-op backward migration for RunPython, and then delete the field). It's unlikely that we'll need to do that in production. Perhaps more likely locally if you need to migrate backward before switching to a different branch. Figured it was worth considering at least.

Reply:
Adding it's quick enough, so I've done so, although migrating backwards in production would be destructive if we had a plan with both regexes populated.